### DISCUSSION TITLE
HEAD -> master in git log? [Beginner]

### DISCUSSION THREAD

**Discussion by Nicknaem** [community/community#22716] [New to GitHub]
<p>Question:<br>
<strong>should (HEAD-&gt;master) message be shown at the top with latest commit from</strong><br>
<em>git log --oneline command</em>,<br>
<strong>does that message only tell on which branch Im currently at?</strong><br>
Hello there :raised_hand_with_fingers_splayed:, I have just begun studing git versrion control, I watched some old tutorials about git basics and there I didn’t see any message like that while showing commits list (when using git log), now when I made several commits I saw that (HEAD -&gt; master) message so I was curious what that meant and as I learned HEAD is just something like ref to another ref which is applied to the latest commit in branch, am I right? so what is the point of showing that message, just to tell on which branch Im currently at?<br>
I achieved hiding that message by using <strong>decorate=no</strong> with <strong>git log</strong> command, could I make that default for git log?, as I saw in  documentation git log uses decoration=short by default and there is mentioned that decorate=short hides refs, ( if I understood it correctly) but it didn’t do that. why? :sweat_smile:</p>
<p>If anyone wants to know what the hell HEAD tag is and why does it point to the master just look this up: <a href="https://people.gnome.org/~federico/news-2008-11.html" rel="nofollow noopener">https://people.gnome.org/~federico/news-2008-11.html</a></p>
<p>any comments would be appreciated I’m just reeal beginner at the moment</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/1/16722b6504181030ef5a91e608ce9fbf7b615c2a.jpeg" data-download-href="/uploads/short-url/3cz6Ud4Ll8xbrsEaB5F6LnrOmZc.jpeg?dl=1" title="GitHelp" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/55950404/181072693-7738f14f-8f11-4fc2-997e-6f14b503887d.jpeg" alt="GitHelp" data-base62-sha1="3cz6Ud4Ll8xbrsEaB5F6LnrOmZc" width="690" height="350" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/1/16722b6504181030ef5a91e608ce9fbf7b615c2a_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">GitHelp</span><span class="informations">826×420 111 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="Nika-m" data-post="1" data-topic="126894">
<div class="title">
<div class="quote-controls"></div>
 Nika-m:</div>
<blockquote>
<p>HEAD is just something like ref to another ref which is applied to the latest commit in branch, am I right?</p>
</blockquote>
</aside>
<p>Most of the time HEAD points at the latest commit in a branch, yes, but it can also point at a plain commit, if you checkout one without creating a branch. That’s called “detached HEAD”.</p>
<aside class="quote no-group" data-username="Nika-m" data-post="1" data-topic="126894">
<div class="title">
<div class="quote-controls"></div>
 Nika-m:</div>
<blockquote>
<p>I achieved hiding that message by using <strong>decorate=no</strong> with <strong>git log</strong> command, could I make that default for git log?</p>
</blockquote>
</aside>
<p>Yes. Set the <code>log.decorate</code> option to <code>no</code>, either using <a href="https://git-scm.com/docs/git-config"><code>git config</code></a> or by editing the configuration directly.</p>
<aside class="quote no-group" data-username="Nika-m" data-post="1" data-topic="126894">
<div class="title">
<div class="quote-controls"></div>
 Nika-m:</div>
<blockquote>
<p>I saw in documentation git log uses decoration=short by default and there is mentioned that decorate=short hides refs, ( if I understood it correctly) but it didn’t do that. why?</p>
</blockquote>
</aside>
<p>Because it doesn’t hide refs, it hides the <code>refs/TYPE/</code> prefix for the refs that are shown. For example, with <code>full</code> instead of <code>short</code> you’d see <code>refs/heads/master</code> instead of <code>master</code>.</p>
<p>In general I recommend you take a look at the <a href="https://git-scm.com/book/en/v2">Git Book</a>. :wink:</p>

**Comment by Nicknaem** [comment]
<p>Thank you so much for answers</p>
